ONSdigital/eq-author-app
# Ear 1800 page titles

### What is the context of this PR?

Add new component to allow for the addition of new required fields to Section, Question, List Collector (3 fields), Confirmation Question, and Calculated Summary.
Adds additional functionality to collapsible to fix display issues with inconsistent use of margins
Add new component for output of inline/in text external links
Adds a new migration to ensure that the new fields show as mandatory when existing questionnaires are opened or imports made.
Note: position of new field has changed from Jira ticket in discussion with Lauren. New field now appears directly after existing title field.

Requires matching eq-publisher-v3 changes to format and output field values for EQ

JIRA: https://jira.ons.gov.uk/browse/EAR-1800
Publisher PR: https://github.com/ONSdigital/eq-publisher-v3/pull/133

### How to review

1. New Questionnaire (tests new fields)

- Create a new questionnaire
- Create a new section, ensuring that the new field shows in the correct location and method, is marked as required on the screen when empty, and contributes to the count of errors for the page correctly. Ensure entered text is stored, retrieved, and in the export. Ensure 'block' is displayed correctly for spacing and position
- Repeat for question
- Repeat for calculated summary
- Repeat for confirmation question
- Repeat for list collector, noting that this has 3 instances of page title, one for each section. Ensure all 3 behave correctly, and that they store and output their values independently

2. Existing questionnaire (test new migration)

- Open an existing questionnaire, and ensure that the new fields are marked as required and contribute to the error counts

3. Importing questions/sections (test new migration)

- Import sections and questions from a non-updated questionnaire, and ensure that they also immediately trigger the required state, and contribute to the error count

4. Test new 'block'

- Ensure text in new block matches spec
- Ensure external link displays with icon, and opens in a new tab when clicked

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
6. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
7. - [ ] Click the **Merge** button on this pull request
8. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
9. - [ ] Move the Jira ticket for this task into the next stage of the process
